src/main/java/taskList/Parser.java

```diff
@@ -0,0 +1,97 @@
+package tasklist;
+
+import tasklist.commands.*;
+
+public class Parser {
+    public enum Commands {
+        MARK,
+        UNMARK,
+        TODO,
+        DEADLINE,
+        EVENT,
+        DELETE,
+        LIST,
+        BYE,
+        HELPG,
+        ADD,
+        UNKNOWN, FIND
+    }
+
+    private static Commands getCommand(String commandString) {
+        try {
+            return Commands.valueOf(commandString.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            return Commands.UNKNOWN;
+        }
+    }
+
+    public static Command parseCommand(String userInput) {
+        String[] inputList = userInput.split(" ", 2);
+        if (inputList.length != 1) { // indicates one of the other commands
+            switch (getCommand(inputList[0])) {
+			case FIND: 
+				return new FindCommand(inputList[1]);
+				
+            case MARK:
+                try {
+                    return new MarkCommand(Integer.parseInt(inputList[1]), true);
+                } catch (NumberFormatException e) {
+                    return new InvalidCommand("Error: Input is not a valid numeric value.");
+                }
+
+            case UNMARK:
+                try {
+                    return new MarkCommand(Integer.parseInt(inputList[1]), false);
+                } catch (NumberFormatException e) {
+                    return new InvalidCommand("Error: Input is not a valid numeric value.");
+                }
+
+            case TODO:
+                return new AddCommand(inputList[1], 1);
+
+            case DEADLINE:
+                return new AddCommand(inputList[1], 2);
+
+            case EVENT:
+                return new AddCommand(inputList[1], 3);
+
+            case DELETE:
+                try {
+                    return new DeleteCommand(Integer.parseInt(inputList[1]));
+                } catch (NumberFormatException e) {
+                    return new InvalidCommand("Error: Input is not a valid numeric value.");
+                }
+
+            case ADD:
+                return new AddCommand(inputList[1], 0);
+
+            case BYE:
+            case LIST:
```

Comment:
Can be removed since `bye` and `list` commands with additional text following them would be handled by the default clause